### ISSUE TITLE
Possible Error in updating Gradient Bandit algorithm

### ISSUE THREAD

**Issue by cbrom** [ShangtongZhang/reinforcement-learning-an-introduction#64]
## Chapter02
In takeAction method updating qEst may be having a problem. 
for selected action (1 - π), and
and for other actions simply π should be applied
But the following code updates it to 
for selected action( 1- π), and
for other actions (0 - π)
`self.qEst = self.qEst + self.stepSize * (reward - baseline) * (oneHot - self.actionProb)`

**Comment by ShangtongZhang** [comment]
No. You can compute the derivation through softmax to check that the implementation is right. It's same with cross-entropy

**Comment by cbrom** [comment]
to be clear, how is 0 - actionProb justified?

**Comment by ShangtongZhang** [comment]
See equation 2.10 in the latest book if you don't want to derive it yourself
![image](https://user-images.githubusercontent.com/6715118/36551087-0a904efc-17b4-11e8-994d-d239b58d8e8e.png)
Note the second equation is minus, so it's equal to (0 - pi)

**Comment by cbrom** [comment]
my bad, Thanks
